src/goto-programs/goto_model.h

```diff
@@ -95,6 +97,29 @@ class goto_modelt : public abstract_goto_modelt
     return goto_functions.function_map.find(id) !=
            goto_functions.function_map.end();
   }
+
+  /// Iterates over the symbols in this goto-model and checks that each one is
+  /// present in the symbol table.
+  /// \return true if no inconsistency was found
+  bool check_id_consistency_symbol_table() const
+  {
+    std::vector<std::string> message_collector;
+    forall_goto_functions(it, goto_functions)
+    {
+      it->second.check_id_consistency_symbol_table(
+        symbol_table, message_collector);
+    }
+    if(!message_collector.empty())
+    {
+      std::cerr << "Error messages collected while checking the goto "
```

Comment:
Also, can you see if you have a reference to a message handler like `error()` or `info()`, instead of writing to `std::cerr`? If it's possible please do that instead.